File: 1160.cpp
#include <iostream>
#include <iomanip>
using namespace std;
int main(){
    int n;
    cin >> n;
    for(int i = 0; i < n; i++){
        int A, B, anos;
        double perA, perB;
        cin >> A >> B >> perA >> perB;
        anos = 0;
        while(true){
            A += A*perA/100;
            B += B*perB/100;
            anos += 1;
            if(anos > 100){
                cout << "Mais de 1 seculo." << endl;
                break;
            }
            if(A > B){
                cout << anos << " anos." << "\n";
                break;
            }
        }
    }
}

File: 1160.java
import java.util.Scanner;
public class Main {
        public static void main(String[] args) {
                Scanner input = new Scanner(System.in);
                int n = input.nextInt();
                int a, b, ano;
                double a1, b1;
                for(int i = 0; i < n; i++){
                        ano = 0;
                        a = input.nextInt(); b = input.nextInt(); a1 = Double.parseDouble(input.next()); b1 = Double.parseDouble(input.next());
                        while (true) {
                                a += (a1 / 100.0 * a);
                                b += (b1 / 100.0 * b);
                                ano+=1;
                                if(ano > 100){
                                        System.out.println("Mais de 1 seculo.");
                                        break;
                                }
                                else if(a > b){
                                        System.out.println(ano+" anos.");
                                        break;
                                }
                        }
                }
        }
}
